assembly/std/index.ts

```diff
@@ -316,10 +316,30 @@ export function unsafeRandom(): i64 {
 /**
  * Sends an async message to a function at given address.
  *
+ * @remarks
+ * The goal of send message is to send a message in the future that will be executed as soon as possible
+ * after the start period but not after the end period. This message allow you to make executions in the future and they
+ * are executed deterministically on all nodes. The execution is made "as soon as possible" because there is a priority
+ * on messages and a limit of messages possibly executed on each slot so if you send a low amount of `rawFee` then your
+ * message could possibly not executed directly at the first slot of the slot period.
+ * If all of the slots of your period have a huge load of messages with more fee than you then it's possible that your
+ * message is never executed.
+ *
+ * Also there is an optional filter on message that add a new condition on the trigger instead of:
+ * "as soon as possible in this range" it becomes
+ * "as soon as possible after this field has been update in the state in this range".
+ *
+ * So you can pass as parameter a `filterAddress` and also an optional datastore `filterKey`.
+ *
+ * If you pass only an address then the message will be executed only after:
+ * "we are in the range and a change has happens during this range on the `filterAddress`" (possibly balance etc).
+ * If you provide a `filterKey` the condition of the execution of the message is:
+ * "we are in the range and a change has happens during this range on the `filterAddress` at that datastore `filterKey`"
+ *
```

Comment:
```suggestion
 * The goal of sendMessage functionality is to send a message in the future, that will be executed as soon as possible
 * after the start period but not after the end period. This message allows you to make executions in the future and they
 * are executed deterministically on all nodes. The execution is made "as soon as possible" because there is a priority
 * on messages and a limit of messages possibly executed on each slot. More precisely, if you send a low amount of `rawFee` then your
 * message could possibly not executed directly at the first slot of the slot period.
 * If all of the slots of the specified period have a large load of messages, with more fees than you specified, then it's highly likely that your
 * message will never be executed.
 *
 * Additionally, there is an optional filter on a message that adds a new condition on the trigger instead of:
 * "as soon as possible in this range", it becomes
 * "as soon as possible, after this field has been updated, in the state, in this range".
 *
 * As a parameter, you can pass the `filterAddress`, and also an optional datastore `filterKey`.
 *
 * If you pass only an address, then the message will be executed only after:
 * "we are in the range, and a change has happened during this range on the `filterAddress`" (possibly balance etc).
 * If you provide a `filterKey`, the condition of the execution of the message is:
 * "we are in the range, and a change has happened during this range on the `filterAddress` at that datastore `filterKey`"
 *
```

Reply:
I have fixed the language but the whole paragraph needs more clarification. I suggest a small call so that we specify how to properly introduce a usecase or a few of them, before we start technical description of how it works

Reply:
cc @AurelienFT

Reply:
Some usecases : 

- Having a smart contract called periodically without a centralized bot
- Having a smart contract that trigger on a value change of an other one (for example a price change)
- Having a object that evolve on the blockchain himself

Reply:
Cool, thanks it will improve this section @AurelienFT .